src/request/query_param.rs

```diff
@@ -1,13 +1,12 @@
 use http::StatusCode;
 use percent_encoding::percent_decode_str;
-use smallvec::SmallVec;
 use std::{borrow::Cow, slice::Iter, str::FromStr};
 
 use crate::{Error, Result};
 
 pub(crate) type QueryParams<T = String> = Vec<(T, (usize, usize))>;
 
-type ValuesVec<'a> = SmallVec<[&'a (usize, usize); 4]>;
+type ValuesVec<'a> = Vec<&'a (usize, usize)>;
```

Comment:
I initially thought that `SmallVec` would internally fall back to using `Vec` from the standard library. However, this is not the case, and I prefer to use `Vec` since we are storing unbounded user input. I'll remain open to potential alternatives in the future, considering something like the following code snippet if it offers significant performance improvements.

```rust
enum FastVec<T, const N: usize> {
    Stack([Option<T>; N]),
    Heap(Vec<T>),
}
```